src/transaction_builder.h

```diff
@@ -52,6 +52,21 @@ struct TransparentInputInfo {
         CAmount value) : scriptPubKey(scriptPubKey), value(value) {}
 };
 
+class TransactionBuilderResult {
+private:
+    boost::optional<CTransaction> maybeTx;
+    boost::optional<std::string> maybeError;
+public:
+    TransactionBuilderResult() = delete;
+    TransactionBuilderResult(const CTransaction& tx);
+    TransactionBuilderResult(const std::string& error);
+    // Always use isTx() or isError() before calling getTxOrThrow() or getError()
```

Comment:
Actually, I think the point of this is semi-related to your other comment. I had originally intended us to call `getError()` (which returns a string) to be rethrown as a different error if `IsError()` was true, and then have `getTxOrThrow()` return the more generic runtime error because we were meant to have rethrown it already. That being said It looks like I made the mistake of not doing it this way when updating z_mergetoaddress. I am just going to make `getTxOrThrow()` throw a `JSONRPCError(RPC_WALLET_ERROR, ...)` along the same lines as making `AddTransparentOutput(...)` (and so on) throw `JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, ... )` rather than runtime errors.